react-components/src/components/RevealToolbar/LayersButton/components/ModelLayerSelection.tsx

```diff
@@ -0,0 +1,55 @@
+/*!
+ * Copyright 2025 Cognite AS
+ */
+
+import { SelectPanel } from '@cognite/cogs-lab';
+import { IconWrapper, ChevronRightSmallIcon } from '@cognite/cogs.js';
+import { type ReactElement, useCallback } from 'react';
+import { type ModelHandler } from '../ModelHandler';
+import { ModelLayersList } from './ModelLayersList';
+import { WholeLayerVisibilitySelectItem } from './WholeLayerVisibilitySelectItem';
+
+export const ModelLayerSelection = ({
+  label,
+  modelLayerHandlers,
+  update
+}: {
+  label: string;
+  modelLayerHandlers: ModelHandler[];
+  update: () => void;
+}): ReactElement => {
```

Comment:
Same here. I find it hard to read. Can you make a separate `ModelLayerSelectionProps` type?